confluent_kafka/avro/serializer/message_serializer.py

```diff
@@ -59,6 +59,30 @@ def __exit__(self, *args):
         return False
 
 
+class HasSchemaMixin:
+    """
+    Wraps a decoded Avro message dictionary to make able to add schema
+    """
+    def schema(self):
+        return self._schema
+
+
+def _wrap(value, schema):
+    if hasattr(schema, 'namespace'):
+        name = "{namespace}.{name}".format(namespace=schema.namespace,
+                                           name=schema.name)
+    elif hasattr(schema, 'name'):
+        name = schema.name
+    else:
+        name = schema.type
+
+    klass = type(str(name), (value.__class__, HasSchemaMixin), {})
+
+    wrapped = klass(value)
+    wrapped._schema = schema
```

Comment:
In the unlikely, but possible, event that existing user's class has a ._schema field, this will silently overwrite it, and is as such a breaking change.

Reply:
I agree, but I do not have a better solution. An option (though a breaking change) would be returning a tuple `(value, schema)` instead, does it make sense?